### PR TITLE
chore: invalid signing time prompt improvement

### PR DESCRIPTION
**Pull request by fanndu**
fix: https://github.com/notaryproject/notation/issues/829

Display the validity period of the certificate when the signing time is invalid.
`./bin/notation sign localhost:5000/gateway@sha256:b992672d71a62c0a94cd8640f7c0db62ffb9de65317100742bf44892b179445f Error: certificate-chain is invalid, certificate with subject "CN=test,O=Notary,L=Seattle,ST=WA,C=US" was invalid at signing time of 2023-11-12 08:33:33 +0000 UTC. Valid signing time from [2023-10-25 02:40:40 +0000 UTC] to [2023-10-26 02:40:40 +0000 UTC]`

### CODE REVIEW

**Review by shizhMSFT — DISMISSED** [review]
LGTM

File: x509/cert_validations.go [hunk 1]
```diff
@@ -120,6 +120,14 @@ func isIssuedBy(subject *x509.Certificate, issuer *x509.Certificate) (bool, erro
 	return bytes.Equal(issuer.RawSubject, subject.RawIssuer), nil
 }
 
+func validateSignTime(cert *x509.Certificate, signingTime *time.Time) error {
+	if signingTime != nil && (signingTime.Before(cert.NotBefore) || signingTime.After(cert.NotAfter)) {
+		return fmt.Errorf("certificate with subject %q was invalid at signing time of %s. Valid signing time from [%s] to [%s]",
```
Comment (priteshbandi) [inline x509/cert_validations.go:125]:
```suggestion
		return fmt.Errorf("certificate with subject %q was invalid at signing time of %s. Certificate is valid from [%s] to [%s]",
```

Comment (fanndu) [reply x509/cert_validations.go]: fix it

**Review by priteshbandi — APPROVED** [review]
LGTM

**Comment by yizha1** [comment]
@fanndu this branch is out-of-date with the base branch. Maybe you need to fix it.

File: x509/cert_validations.go [hunk 2]
```diff
@@ -120,6 +120,14 @@ func isIssuedBy(subject *x509.Certificate, issuer *x509.Certificate) (bool, erro
 	return bytes.Equal(issuer.RawSubject, subject.RawIssuer), nil
 }
 
+func validateSignTime(cert *x509.Certificate, signingTime *time.Time) error {
```
Comment (Two-Hearts) [inline x509/cert_validations.go:123]: nit:
```suggestion
func validateSigningTime(cert *x509.Certificate, signingTime *time.Time) error {
```

**Comment by fanndu** [comment]
> @fanndu this branch is out-of-date with the base branch. Maybe you need to fix it.

File: x509/cert_validations.go (hunk 2 above)
Comment (fanndu) [reply x509/cert_validations.go]: Fix it

**Comment by fanndu** [comment]
> @fanndu this branch is out-of-date with the base branch. Maybe you need to fix it.

Thanks. Already fix it.

**Comment by yizha1** [comment]
@priteshbandi @patrickzheng200 Would you mind reviewing this PR again? Thanks.

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Comment by fanndu** [comment]
> @fanndu I'm trying to merge this PR. Could you rebase the branch?

Done. Thanks.

**Pull request merged**